config/initializers/fix_neo4j_wait_for_connection.rb

```diff
@@ -0,0 +1,31 @@
+module Neo4j
+  class SessionManager
+    class << self
+      def open_neo4j_session(type, url_or_path, wait_for_connection = false, options = {})
+        enable_unlimited_strength_crypto! if java_platform? && session_type_is_embedded?(type)
+
+        adaptor = cypher_session_adaptor(type, url_or_path, options.merge(wrap_level: :proc))
+        session = Neo4j::Core::CypherSession.new(adaptor)
+        wait_and_retry(session) if wait_for_connection
+        session
+      end
+
+      protected
+
+      def wait_and_retry(session)
+        logger = Neo4j::Config[:logger]
+        Timeout.timeout(60) do
+          begin
+            logger.fatal ?.
+            session.constraints
+          rescue Neo4j::Core::CypherSession::ConnectionFailedError
+            sleep(1)
```

Comment:
Is the `sleep` needed here?

Reply:
> Is the `sleep` needed here?

It's there to prevent flooding the server. It was on the previous code and I didn't actually try to remove, but I'm pretty sure there will be as much flood as possible.

Reply:
cool tnx